R/tbl_survfit_quantiles.R

```diff
@@ -0,0 +1,245 @@
+#'     statistic = everything() ~ cards::continuous_summary_fns(c("min", "max"))
+#'   )
+#'
+#' # stack the ARDs and pass them to `tbl_ard_summary()`
+#' cards::bind_ard(
+#'   ard_surv_quantiles,
+#'   ard_surv_min_max
+#' ) |>
+#'   tbl_ard_summary(
+#'     by = "TRTA",
+#'     type = list(prob = "continuous2", AVAL = "continuous"),
+#'     statistic = list(
+#'       prob = c("{estimate50}", "({conf.low50}, {conf.high50})", "{estimate25}, {estimate75}"),
+#'       AVAL = "{min} to {max}"
+#'     ),
+#'     label = list(
+#'       prob = "Time to event",
+#'       AVAL = "Range"
+#'     )
+#'   ) |>
+#'   # directly modify the labels in the table to match spec
+#'   modify_table_body(
+#'     ~ .x |>
+#'       mutate(
+#'         label = dplyr::case_when(
+#'           .data$label == "Survival Probability" ~ "Median",
+#'           .data$label == "(CI Lower Bound, CI Upper Bound)" ~ "95% CI",
+#'           .data$label == "Survival Probability, Survival Probability" ~ "25% and 75%-ile",
+#'           .default = .data$label
+#'         )
+#'       )
+#'   ) |>
+#'   # update indentation to match spec
+#'   modify_column_indent(columns = "label", rows = label == "95% CI", indent = 8L) |>
+#'   modify_column_indent(columns = "label", rows = .data$label == "Range", indent = 4L) |>
+#'   # remove default footnotes
+#'   remove_footnote_header(columns = all_stat_cols())
+#' ```
+#'
+#' @examples
+#' tbl_survfit_quantiles(
+#'   data = cards::ADTTE,
+#'   by = "TRTA"
+#' )
+tbl_survfit_quantiles <- function(data,
+                                  y = "Surv_CNSR()",
+                                  by = NULL,
+                                  header = "Time to event",
+                                  estimate_fun = label_style_number(digits = 1),
+                                  method.args = list(conf.int = 0.95)) {
+  method.args <- enquo(method.args)
+
+  # check inputs ---------------------------------------------------------------
+  set_cli_abort_call()
+  check_not_missing(data)
+  check_string(y)
+  check_string(header)
+  cards::process_selectors(data, by = {{ by }})
+  check_class(estimate_fun, "function")
+  if (length(by) > 1L) {
+    cli::cli_abort(
+      "The {.arg by} argument must be empty or a single stratifying variable name.",
+      call = get_cli_abort_call()
+    )
+  }
+  if ("time" %in% by) {
+    cli::cli_abort(
+      "The {.arg by} column cannot be named {.val time}.",
+      call = get_cli_abort_call()
+    )
+  }
+  func_inputs <- as.list(environment())
+
+  # subset data on complete row ------------------------------------------------
+  form <- glue("{y} ~ {ifelse(is_empty(by), 1, cardx::bt(by))}")
+  data <- data[stats::complete.cases(data[all.vars(form)]), ]
+
+  # get survival quantiles -----------------------------------------------------
+  ard_surv_quantiles <-
+    cardx::ard_survival_survfit(
+      x = data,
+      y = y,
+      variables = any_of(by),
+      probs = c(0.25, 0.50, 0.75),
```

Comment:
why not surfacing this default?

Reply:
(ARD-first made this clear but I am still wondering if we could select more paramenters that can be useful to have also here)

Reply:
yeah, this table is just here because it's a common structure in the templates. The ARD-first code is provided in case they need to make any changes.